template_tag.go

```diff
@@ -0,0 +1,84 @@
+package faker
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strings"
+	"text/template"
+)
+
+// EvaluateTemplateFields evaluates the template-tagged fields for a struct value `v` of type `t`.
+// tagName is the field tag name to use (typically opts.TagName).
+func EvaluateTemplateFields(t reflect.Type, v reflect.Value, templateFields []int, tagName string) error {
+	if len(templateFields) == 0 {
+		return nil
+	}
+	// build context map from current struct values
+	ctx := make(map[string]interface{})
+	for i := 0; i < v.NumField(); i++ {
+		ctx[t.Field(i).Name] = v.Field(i).Interface()
+	}
+
+	for _, idx := range templateFields {
+		tags := decodeTags(t, idx, tagName)
+		tpl := tags.fieldType
+		lower := strings.ToLower(tpl)
+		if strings.HasPrefix(lower, TemplateTag+":") {
+			tpl = tpl[len(TemplateTag)+1:]
+		}
+		res, err := evaluateTemplateForField(tpl, ctx)
+		if err != nil {
+			return err
+		}
+		if v.Field(idx).Kind() != reflect.String {
+			return fmt.Errorf("template tag only supported on string fields: %s", t.Field(idx).Name)
+		}
+		v.Field(idx).SetString(res)
+	}
+	return nil
+}
+
+// evaluateTemplateForField evaluates a template string tpl with context ctx (map of field name -> value)
+// and returns the resulting string. Only a small set of helper funcs are exposed.
+func evaluateTemplateForField(tpl string, ctx map[string]interface{}) (string, error) {
```

Comment:
We can modernize by using `any`
```suggestion
func evaluateTemplateForField(tpl string, ctx map[string]any) (string, error) {
```